android/WSL.README.md

```diff
@@ -0,0 +1,76 @@
+# Building BOINC Android client on Windows (WSL)
+
+## Goal
+
+Allow Windows users to use built-in WSL for setting up for BOINC Android development
+
+## Limitations
+
+* Windows 10 only
+* Windows Defender may affect compile times (see [Tips#1](#Tips))
+* fork/exec error on WSL (see [Tips#1](#Tips))
+
+## Prerequisites
+
+* Windows 10
+* [Android Studio (Windows)](https://developer.android.com/studio#downloads)
+* [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install-win10) # IMPORTANT #
+* [Ubuntu from Microsoft Store](https://wiki.ubuntu.com/WSL#Installing_Ubuntu_on_WSL_via_the_Microsoft_Store_.28Recommended.29)
+
+## Set up Ubuntu
+
+We are going to use Ubuntu 18.04 as an example, other versions should be similar. Refer [Prerequisites](#Prerequisites) before proceeding.
+
+1. Update and install build tools
+
+        $ sudo apt update && sudo apt upgrade
+
+        $ sudo apt install git unzip make m4 pkg-config autoconf automake libtool python
+
+1. Clone BOINC repo
+
+        $ git clone https://github.com/boinc/boinc.git ~/boinc
+
+1. Build BOINC Android client
+
+        $ cd ~/boinc/android
+
+        $ ./build_all.sh
+
+    You should see this once you are done compiling:
+
+        ===== BOINC for all platforms build done =====
+
+    If you ever had problem at this point, try `git clean -fxd ..` and run again
+
+1. Copy the main BOINC repo to Android Studio readable path. Replace `<pathtofolder>` with appropriate location.
+
+        $ cp -fr ~/boinc /mnt/c/<pathtofolder>/boinc
```

Comment:
You should warn users here that after moving files to Windows accessible location there will be a mess with line-endings (if they want to continue development on Windows, line-ending should be adjusted manually). As an alternative ,yo can suggest users not to move the whole repository to Windows accessible location but only assets files (from `android\BOINC\app\src\main\assets` folder)

Reply:
This could interfere with APK build because it relies on git tags, I see what I can do

Reply:
Both systems can checkout the same commit. Basically, you can run vscode locally with connection to WSL, install gradle plugin and work with Java code without Android Studio

Reply:
Can we settle on whether we want to use Android Studio or VS Code + various plugins to develop for Android? I don't intend to introduce another IDE just for WSL. My current workflow is commit changes on WSL and test the code in Android Studio

Reply:
Again, it's better to have two different source code bases, one inside WSL, another outside of it.
On WSL you can build native binaries and move only them to Windows and continue development on another source base with Android Studio

Reply:
Ok that's acceptable